app/models/talk.rb

```diff
@@ -60,8 +60,12 @@ class Talk < ApplicationRecord
   # delegates
   delegate :name, to: :event, prefix: true, allow_nil: true
 
+  # callbacks
+  before_validation :set_kind
```

Comment:
Would it make sense to only run this when the title changed as well?
```suggestion
  before_validation :set_kind, if: :title_changed?
```